src/util/json_expr.cpp

```diff
@@ -260,18 +274,33 @@ json_objectt json(
 
       bool is_signed=underlying_type.id()==ID_signedbv;
 
-      std::string sig=is_signed?"":"unsigned ";
-
-      if(width==config.ansi_c.char_width)
-        result["c_type"]=json_stringt(sig+"char");
-      else if(width==config.ansi_c.int_width)
-        result["c_type"]=json_stringt(sig+"int");
-      else if(width==config.ansi_c.short_int_width)
-        result["c_type"]=json_stringt(sig+"short int");
-      else if(width==config.ansi_c.long_int_width)
-        result["c_type"]=json_stringt(sig+"long int");
-      else if(width==config.ansi_c.long_long_int_width)
-        result["c_type"]=json_stringt(sig+"long long int");
+      if(mode==ID_C)
```

Comment:
Why not use our existing pretty-printers here? For example, the way show-symbol-table pretty-prints types (e.g. https://github.com/diffblue/cbmc/blob/d4e8149fb5df3e7906a9d6b133c89033797cd4ab/src/goto-programs/show_symbol_table.cpp#L70)